src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.Patch.cs

```diff
@@ -208,14 +208,15 @@ private object AddTypeNameToValueIfNeeded(Type propertyType, object value)
             if (propertyType == typeOfValue || typeOfValue.IsClass == false)
                 return value;
 
-            using (var writer = new BlittableJsonWriter(Context))
+            using (var writer = Conventions.Serialization.CreateWriter(Context))
             {
-                // the type of the object that's being serialized 
+                // the type of the object that's being serialized
                 // is not the same as its declared type.
                 // so we need to include $type in json
 
-                var serializer = Conventions.CreateSerializer();
-                serializer.TypeNameHandling = TypeNameHandling.Objects;
+                var serializer = Conventions.Serialization.CreateSerializer();
+                var jsonSerializer = (JsonNetJsonSerializer)serializer;
```

Comment:
I will check

Reply:
added CreateSerializerOptions here with our own TypeNameHandling. This could be a problem when implementing a different serializer, but not sure if we can avoid this